src/execution/mod.rs

```diff
@@ -21,24 +21,37 @@ mod stats;
 use std::io::Write;
 use std::path::PathBuf;
 use std::sync::Arc;
+use std::time::Duration;
 
 use log::{error, info};
 pub use stats::{collect_plan_stats, ExecutionStats};
 
 use crate::config::ExecutionConfig;
 use crate::extensions::{enabled_extensions, DftSessionStateBuilder};
-use color_eyre::eyre::Result;
+use color_eyre::eyre::{self, Result};
 use datafusion::execution::SendableRecordBatchStream;
-use datafusion::physical_plan::ExecutionPlan;
+use datafusion::physical_plan::{execute_stream, ExecutionPlan};
 use datafusion::prelude::*;
-use datafusion::sql::parser::Statement;
+use datafusion::sql::parser::{DFParser, Statement};
 use tokio_stream::StreamExt;
 #[cfg(feature = "flightsql")]
 use {
     crate::config::FlightSQLConfig, arrow_flight::sql::client::FlightSqlServiceClient,
     tokio::sync::Mutex, tonic::transport::Channel,
 };
 
+// Container for benchmark statistics for a single location execution
+//
+// Note: It would be nice if we could split out planning and optimization for both logical and
+// physical planning but its not obvious from the DataFusion APIs how to do that.
+#[derive(Debug)]
+pub struct LocalBenchmarkStats {
+    logical_planning_duration: Duration,
+    physical_planning_duration: Duration,
+    execution_duration: Duration,
+    total_duration: Duration,
+}
+
```

Comment:
maybe ill change this so that each metric is a vec, convert to arrow array, and then use arrow kernel for summary stats

Reply:
or...maybe i can create mem table and insert values to that and just run some sql to get summary stats...

Reply:
realistically ill do something simple to start and consider either of the above an optimization / improvement